src/components/Carousel/Carousel.stories.tsx

```diff
@@ -94,19 +98,153 @@ const Scroll_Story: ComponentStory<typeof Carousel> = (args) => (
   <Carousel {...args} />
 );
 
+type scrollVisibilityApiType = React.ContextType<typeof VisibilityContext>;
+
+// When using custom buttons, manually handle touch via scrollVisibilityApiType and apiRef.
+const onWheel = (
+  apiObj: scrollVisibilityApiType,
+  event: React.WheelEvent
+): void => {
+  const isTouchpad =
+    Math.abs(event?.deltaX) !== 0 || Math.abs(event?.deltaY) < 15;
+
+  if (isTouchpad) {
+    event?.stopPropagation();
+    return;
+  }
+
+  if (event?.deltaY < 0) {
+    apiObj?.scrollNext();
+  } else if (event.deltaY > 0) {
+    apiObj?.scrollPrev();
+  }
+};
+
+const Scroll_Custom_Buttons_Story: ComponentStory<typeof Carousel> = (args) => {
+  const htmlDir: string = useCanvasDirection();
+  const apiRef: React.MutableRefObject<autoScrollApiType> = useRef(
+    {} as scrollVisibilityApiType
+  );
+
+  const [nextDisabled, setNextDisabled] = useState<boolean>(false);
+  const [previousDisabled, setPreviousDisabled] = useState<boolean>(true);
+
+  const scrollPrevAsync = async (): Promise<void> => {
+    await apiRef.current?.scrollPrev();
+    setNextDisabled(apiRef.current.isLastItemVisible);
+    setPreviousDisabled(apiRef.current.isFirstItemVisible);
+  };
+
+  const scrollNextAsync = async (): Promise<void> => {
+    await apiRef.current?.scrollNext();
+    setNextDisabled(apiRef.current.isLastItemVisible);
+    setPreviousDisabled(apiRef.current.isFirstItemVisible);
+  };
+
+  return (
+    <Card bordered height={440} style={{ padding: 48 }} width={400}>
+      <div
+        style={{
+          height: '100%',
+          left: htmlDir === 'rtl' ? 'unset' : 8,
+          position: 'absolute',
+          right: htmlDir === 'rtl' ? 8 : 'unset',
+          top: 0,
+          width: 28,
+        }}
+      >
+        <Button
+          ariaLabel="Previous"
+          disabled={previousDisabled}
+          iconProps={{
+            path:
+              htmlDir === 'rtl'
+                ? IconName.mdiChevronRight
+                : IconName.mdiChevronLeft,
+          }}
+          onClick={() => {
+            if (apiRef.current?.isFirstItemVisible) {
+              console.log('first item visible');
+              setPreviousDisabled(true);
+              return;
+            }
+            console.log('previous clicked');
+            scrollPrevAsync();
+          }}
+          shape={ButtonShape.Round}
+          size={ButtonSize.Small}
+          style={{
+            position: 'absolute',
+            top: 'calc(50% - 14px)',
+          }}
+          variant={ButtonVariant.Secondary}
+        />
+      </div>
+      <Carousel
```

Comment:
could you also point me to the code where the original carousel buttons are disabled? Which props are controlling that

Reply:
`controls` prop is the one. Here is the code for [previous](https://github.com/EightfoldAI/octuple/blob/main/src/components/Carousel/ScrollMenu/ScrollMenu.tsx#L167) and for [next](https://github.com/EightfoldAI/octuple/blob/main/src/components/Carousel/ScrollMenu/ScrollMenu.tsx#L185). You should see in the storybook args we set `controls` to `false` in order to use our custom buttons (or any other element for that matter as it can be foo).

Reply:
This Story is meant to show how extensible the Octuple `Carousel` actually is with regards to implementation.